Matrix.java

```diff
@@ -0,0 +1,210 @@
+class Matrix
+{
+    final double[][] a;
+    
+    Matrix(double[][] A)
+    {
+	this(A,0,0);
+    }
+    Matrix(double[][] A, int m, int n)
+    {
+        int rows = A.length;
+        m = rows < m ? rows : m;
```

Comment:
@ccampell I don't know what you mean about lines 9-10. Maybe it just worked that way because of where I made comments?